src/Dapr.Extensions.Configuration/DaprSecretDescriptor.cs

```diff
@@ -31,20 +31,51 @@ public class DaprSecretDescriptor
         public IReadOnlyDictionary<string, string> Metadata { get; }
 
         /// <summary>
-        /// Secret Descriptor Construcutor
+        /// This flag indicates if this field's existence should trigger an exception. Setting it to "false"
+        /// will suppress the exception whereas setting it to "true" will not suppress it.
+        /// </summary>
+        public bool IsRequired { get; }
+
+        /// <summary>
+        /// SecretKey value is mapping value to Vault Name. If The application's Secret Name and Secret in the
+        /// Vault Name is different then you can use this flag to specify Vault Secret Name. 
+        /// </summary>
+        public string SecretKey { get; }
+
+        /// <summary>
+        /// Secret Descriptor Constructor
         /// </summary>
         public DaprSecretDescriptor(string secretName) : this(secretName, new Dictionary<string, string>())
         {
 
         }
 
         /// <summary>
-        /// Secret Descriptor Construcutor
+        /// Secret Descriptor Constructor
+        /// </summary>
+        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata) :
+            this(secretName, metadata, true, secretName)
+        {
+
+        }
+
+        /// <summary>
+        /// Secret Descriptor Constructor
+        /// </summary>
+        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata, bool isRequired) :
+            this(secretName, metadata, isRequired, secretName)
+        {
+        }
+
+        /// <summary>
+        /// Secret Descriptor Constructor
         /// </summary>
-        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata)
+        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata, bool isRequired, string secretKey)
```

Comment:
Consider making `isRequired` and `secretKey` both optional arguments, to help reduce the explosion of constructor overloads as well as, say, allow users to *only* specify the `secretKey` but not have to then care about the `isRequired` argument.  We might also consider doing the same for `metadata`.

Ideally these would all (save for `secretName`) bet `init` properties on the class, but that's a larger breaking change (depending on how many people *actually* specify metdata with each secret).

Reply:
Removed the constructor overload bloat and consolidated this change into the existing 2 constructors using optional arguments.